src/pudl/output/ferc1.py

```diff
@@ -1862,87 +2062,84 @@ def forest(self: Self) -> nx.DiGraph:
         table may or may not have a top level summary value that includes all underlying
         calculated values of interest.
         """
-        forest = self.seeded_digraph
-        # Remove any node that has only one parent and one child, and add an edge
-        # between its parent and child.
-        for node in self.passthroughs:
-            parent = list(forest.predecessors(node))
-            assert len(parent) == 1
-            successors = forest.successors(node)
-            assert len(list(successors)) == 2
-            child = [
-                n
-                for n in forest.successors(node)
-                if not n.xbrl_factoid.endswith("_correction")
-            ]
-            correction = [
-                n
-                for n in forest.successors(node)
-                if n.xbrl_factoid.endswith("_correction")
-            ]
-            assert len(child) == 1
-            logger.debug(
-                f"Replacing passthrough node {node} with edge from "
-                f"{parent[0]} to {child[0]}"
-            )
-            forest.remove_nodes_from(correction + [node])
-            forest.add_edge(parent[0], child[0])
-
-        if not nx.is_forest(forest):
-            logger.error(
-                "Calculations in Exploded Metadata can not be represented as a forest!"
-            )
-        connected_components = list(nx.connected_components(forest.to_undirected()))
-        logger.debug(
-            f"Calculation forest contains {len(connected_components)} connected components."
-        )
-
-        # Remove any node that:
-        # - ONLY has stepchildren.
-        # - AND has utility_type total
-        for node in self.stepparents(forest):
+        forest = deepcopy(self.seeded_digraph)
+        # Remove any node that ONLY has stepchildren.
+        pure_stepparents = []
+        stepparents = sorted(self.stepparents(forest))
+        logger.info(f"Investigating {len(stepparents)=}")
+        for node in stepparents:
             children = set(forest.successors(node))
             stepchildren = set(self.stepchildren(forest)).intersection(children)
-            if (
-                (children == stepchildren)
-                & (len(children) > 0)
-                & (node.utility_type == "total")
-            ):
+            if (children == stepchildren) & (len(children) > 0):
+                pure_stepparents.append(node)
                 forest.remove_node(node)
+        logger.info(f"Removed {len(pure_stepparents)} redundant/stepparent nodes.")
```

Comment:
Probably good to define stepparents/stepchildren somewhere 😅

Reply:
There are `calc_forest.stepparents` and `calc_forest.stepchildren` methods / properties that define them but maybe I should reference those methods directly here.